### ISSUE TITLE
Extra documents in ES

### ISSUE THREAD

**Issue by alex-oleshkevich** [rwynn/monstache#74]
This is a separate issue for #67

1. I have a dump of database with 2.2M documents;
2. I import it into mongo (monstache is off);
3. Once import is finished, I start monstache and wait for it to finish;
4. Start monstache in tailing mode;
4. Compare doc counts: ES has more documents than mongo.

Also, when services work for a long time (~a week), the diff count becomes larger.
At moment I have 2281768 docs in mongo and 2281793 in ES (25 in diff).
The count 25 is pretty constant, I do not know what does it relates to.

My app is: a background job receives JSON from RabbitMQ and posts it into mongodb via mongoengine (python ODM). Then monstache picks up the change and index it in the ES.

From our pass discussion:
`That’s interesting about the 2 extra docs. Do you do hard deletes in MongoDB? If so, any chance you override the default setting for index.gc_deletes on your indexes. I believe that defaults to 60s. Monstache may send index and delete requests out of order. But it uses a version number for each document with the timestamp from the oplog. That should prevent certain indexes requests from occurring if the version number is lower than last or if the same doc has been deleted in the last 60s. Just something to look into.`

I am not sure if I have that.

**Comment by alex-oleshkevich** [comment]
I know that this info does not helps a lot, but tell me if I can measure something for you

**Comment by rwynn** [comment]
Also how are you doing the count in Elasticsearch?  Are you counting docs in a single index or wildcard?

Are you indexing stats?  That generates and indexes a document by default every 30s that monstache is running.  It would not be in the index you with your mongo data of course. 

You may want to set namespace-regex option to e.g. ^autobot\.advert$ if you only need that collection. Otherwise, monstache will sync changes detected in other collections. By default it syncs any change detected in the oplog.

**Comment by alex-oleshkevich** [comment]
for mongo: 
```
use autobot
db.advert.count()
```

for ES:
`localhost:9200/autobot/advert/_count`

**Comment by alex-oleshkevich** [comment]
i cleared indices in es and restarted sync -- counts are equal.
in my app documents in mongodb are never deleted and there are no other process writing to es.
current diff is still 25.

i will keep monitoring that and let you know if i find something

**Comment by rwynn** [comment]
Counts are now equal or 25 diff?  Sorry I didn’t understand that last one.

**Comment by alex-oleshkevich** [comment]
I executed sync locally using prod db dump. Numbers are equal.

On prod (I didn't change anything there), the diff is 25 documents.

**Comment by rwynn** [comment]
Ok thanks. Is it possible the oplog in prod could have had data in it already before the data dump imported?
